src/libcore/ops/place.rs

```diff
@@ -27,10 +27,13 @@
 /// implementation of Place to clean up any intermediate state
 /// (e.g. deallocate box storage, pop a stack, etc).
 #[unstable(feature = "placement_new_protocol", issue = "27779")]
-pub trait Place<Data: ?Sized> {
+pub unsafe trait Place<Data: ?Sized> {
     /// Returns the address where the input value will be written.
     /// Note that the data at this address is generally uninitialized,
     /// and thus one should use `ptr::write` for initializing it.
+    ///
+    /// This function must return a valid (non-zero) pointer to
+    /// a location at which a value of type `Data` can be written.
```

Comment:
To stave off confusion then, maybe this should say "must return a valid pointer". Since non-null is not the end of the validity requirements.

Reply:
> the end of the validity requirements

Are those requirements listed somewhere where we can refer to?

Reply:
The wording is actually kind of redundant as-is. The real requirement is "This function must return a pointer that a value of type `Data` can be written through."

Reply:
I agree, it's redundant.